cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/realtime/sinks/StreamSink.java

```diff
@@ -0,0 +1,154 @@
+  }
+
+  @Override
+  public void configurePipeline(ETLStage stageConfig, PipelineConfigurer pipelineConfigurer) {
+    Map<String, String> properties = stageConfig.getProperties();
+    String streamName = properties.get(Properties.Stream.NAME);
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(streamName), "Stream name should be non-null, non-empty.");
+    pipelineConfigurer.addStream(new Stream(streamName));
+  }
+
+  @Override
+  public void initialize(SinkContext context) throws Exception {
+    super.initialize(context);
+    Map<String, String> runtimeArguments = context.getRuntimeArguments();
+    streamName = runtimeArguments.get(Properties.Stream.NAME);
+    headersField = runtimeArguments.containsKey(Properties.Stream.HEADERS_FIELD) ?
+      runtimeArguments.get(Properties.Stream.HEADERS_FIELD) : Properties.Stream.DEFAULT_HEADERS_FIELD;
+    timestampField = runtimeArguments.containsKey(Properties.Stream.TIMESTAMP_FIELD) ?
+      runtimeArguments.get(Properties.Stream.TIMESTAMP_FIELD) : Properties.Stream.DEFAULT_TIMESTAMP_FIELD;
+    dataField = runtimeArguments.containsKey(Properties.Stream.DATA_FIELD) ?
+      runtimeArguments.get(Properties.Stream.DATA_FIELD) : Properties.Stream.DEFAULT_DATA_FIELD;
+  }
+
+  @Override
+  public void write(Iterable<StructuredRecord> structuredRecords) throws Exception {
+    SinkContext context = getContext();
+    for (StructuredRecord structuredRecord : structuredRecords) {
+      Schema schema = structuredRecord.getSchema();
+      Object data = structuredRecord.get(dataField);
+      Object timestamp = structuredRecord.get(timestampField);
+      Object headers = structuredRecord.get(headersField);
+      if (data == null) {
+        LOG.debug("Found null data. Skipping record.");
+        return;
+      }
+      Schema.Field dataSchemaField = schema.getField(dataField);
+      switch (dataSchemaField.getSchema().getType()) {
+        case BYTES:
+          writeBytes(context, headers, timestamp, data);
+          break;
+        case STRING:
+          writeString(context, data, headers);
+          break;
+        default:
+          throw new UnsupportedTypeException(String.format("Type %s is not supported for writing to stream",
+                                                          data.getClass().getName()));
+
+      }
+    }
+  }
+
+  private void writeBytes(SinkContext context, Object headers, Object timestamp,
+                          Object data) throws IOException, UnsupportedTypeException {
+    ByteBuffer buffer;
+    if (data instanceof ByteBuffer) {
+      buffer = (ByteBuffer) data;
+    } else if (data instanceof byte []) {
+      buffer = ByteBuffer.wrap((byte []) data);
+    } else {
+      throw new UnsupportedTypeException(String.format("Type %s is not supported for writing to stream",
+                                                       data.getClass().getName()));
+    }
+    if (headers != null && headers instanceof Map) {
```

Comment:
we can also check the field to map sure its a map<string, string>, though it may not be worth it since it's an error either way